server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/ClusterBalanceStats.java

```diff
@@ -241,14 +243,24 @@ private static NodeBalanceStats createFrom(
                 }
             }
 
+            assert desiredBalance != null;
+            Double nodeWeight = desiredBalance.weightsPerNode().isEmpty()
+                ? null
+                // Desired balance is computed asynchronously so its possible a new node from the latest cluster state
+                // hasn't been included yet
+                : Optional.ofNullable(desiredBalance.weightsPerNode().get(routingNode.node()))
+                    .map(DesiredBalanceMetrics.NodeWeightStats::nodeWeight)
+                    .orElse(null);
```

Comment:
Seems a bit unnecessary to check `isEmpty` and `Optional` separately since the later covers the former. I understand they happen for different reasons. But maybe a comment explaining both situations is sufficient.